ocaml/xxhash/lib/XXHash.ml

```diff
@@ -0,0 +1,94 @@
+(*
+Copyright (c) 2016 Pieter Goetschalckx
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*)
+
+exception XXH_error
+
+module type XXHASH = sig
+  type hash
+  type state
+
+  val hash : ?seed:hash -> string -> hash
+  val create : unit -> state
+  val reset : ?seed:hash -> state -> unit
+  val update : state -> string -> unit
+  val digest : state -> hash
+  val free : state -> unit
+
+  val with_state : ?seed:hash -> (state -> unit) -> hash
```

Comment:
I believe there should be no public way to create state such that `with_state` is the only way to obtain and release state; hence, `create` and `free` should be private and not show up in `XXHASH`.